### PR TITLE
Lint for strict-boolean-expressions

### PR DESCRIPTION
**Pull request by andy-hanson**
#### PR checklist

- [ ] Addresses an existing issue: https://github.com/palantir/tslint/pull/2346#discussion_r108027137
- [ ] New feature, bugfix, or enhancement
  - [ ] Includes tests
- [ ] Documentation update

#### Overview of change:

Applies the `strict-boolean-expressions` rule to the code.

### CODE REVIEW

File: src/formatters/vsoFormatter.ts [hunk 1]
```diff
@@ -43,7 +43,7 @@ export class Formatter extends AbstractFormatter {
             const lineAndCharacter = failure.getStartPosition().getLineAndCharacter();
             const line = lineAndCharacter.line + 1;
             const character = lineAndCharacter.character + 1;
-            const code = (failure.getRuleName ? failure.getRuleName() : "");
+            const code = failure.getRuleName();
```
Comment (adidahiya) [inline src/formatters/vsoFormatter.ts]: nice

File: src/rules/completedDocsRule.ts [hunk 2]
```diff
@@ -433,7 +433,10 @@ class CompletedDocsWalker extends Lint.ProgramAwareRuleWalker {
 
     private describeModifier(kind: ts.SyntaxKind) {
         const description = ts.SyntaxKind[kind].toLowerCase().split("keyword")[0];
-
-        return CompletedDocsWalker.modifierAliases[description] || description;
+        const alias = CompletedDocsWalker.modifierAliases[description];
+        if (alias !== undefined) {
```
Comment (adidahiya) [inline src/rules/completedDocsRule.ts]: nit: would've preferred a ternary expression here

File: src/rules/noConsecutiveBlankLinesRule.ts [hunk 3]
```diff
@@ -53,12 +53,12 @@ export class Rule extends Lint.Rules.AbstractRule {
...
     public isEnabled(): boolean {
         const option = this.ruleArguments[0] as number | undefined;
-        return super.isEnabled() && (!option || option > 0);
+        return super.isEnabled() && (option === undefined || option > 0);
     }
 
     public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
-        const limit = this.ruleArguments[0] as number | undefined || Rule.DEFAULT_ALLOWED_BLANKS;
-        return this.applyWithFunction(sourceFile, walk, limit);
+        const limit = this.ruleArguments[0] as number | undefined ;
```
Comment (adidahiya) [inline src/rules/noConsecutiveBlankLinesRule.ts]: nit: extra space before `;`

File: src/rules/orderedImportsRule.ts [hunk 4]
```diff
@@ -161,12 +161,12 @@ class OrderedImportsWalker extends Lint.RuleWalker {
...
 
-        interface Options { "import-sources-order": string; "named-imports-order": string; }
-        const optionSet = (this.getOptions() as [Options])[0] || {};
-        this.importSourcesOrderTransform =
-            TRANSFORMS[optionSet["import-sources-order"] || "case-insensitive"];
-        this.namedImportsOrderTransform =
-            TRANSFORMS[optionSet["named-imports-order"] || "case-insensitive"];
+        interface Options { "import-sources-order"?: string; "named-imports-order"?: string; }
+        const optionSet = (this.getOptions() as [Options])[0];
+        const { "import-sources-order": sources = "case-insensitive", "named-imports-order": named = "case-insensitive" } =
```
Comment (adidahiya) [inline src/rules/orderedImportsRule.ts]: this is kind of a crazy line. might be nice to spread it out:

```ts
const {
    "import-sources-order": sources = "case-insensitive",
    "named-imports-order": named = "case-insensitive",
} = optionSet === undefined ? {} : optionSet;
```

File: src/rules/strictTypePredicatesRule.ts [hunk 5]
```diff
@@ -207,7 +208,7 @@ function isFunction(t: ts.Type): boolean {
         return true;
     }
     const symbol = t.getSymbol();
-    return (symbol && symbol.getName()) === "Function";
+    return symbol === undefined ? false : symbol.getName() === "Function";
```
Comment (adidahiya) [inline src/rules/strictTypePredicatesRule.ts]: nit: a simple boolean expression would be slightly easier to read, `symbol !== undefined && symbol.getName() === "Function"`

Comment (andy-hanson) [inline src/rules/strictTypePredicatesRule.ts]: #2534 will catch this. :zap:

**Comment by andy-hanson** [comment]
As for running this rule in vscode, see Microsoft/vscode-tslint#70.

**Comment by egamma** [comment]
@andy-hanson @adidahiya there is a preview of the vscode-tslint extension that uses the typescript server plugin which supports type checked rules. All you need to do is to install this [extension](https://marketplace.visualstudio.com/items?itemName=eg2.ts-tslint). 

One thing to keep in mind when using the tslint typescript server plugin is that the `no-unused-variable` rule is disabled because of this tslint issue palantir/tslint#2649.

**Pull request merged**